### PR TITLE
Feat/database row

### PR DESCRIPTION
**Pull request by speed2exe**
- Expose some fields from `DatabaseRowBody` for publish duplication
- Update all `meta` for `DatabaseRowBody` when `row_id` is updated
- Added method `has_document` to `DatabaseRowBody`

### CODE REVIEW

File: collab-database/src/rows/row.rs [hunk 1]
```diff
@@ -213,24 +213,30 @@ impl BorrowMut<Collab> for DatabaseRow {
 
 pub struct DatabaseRowBody {
   row_id: RowId,
-  data: MapRef,
+  pub data: MapRef,
```
Comment (appflowy) [inline collab-database/src/rows/row.rs:216]: can we keep it private by using method/functions instead

File: collab-database/src/rows/row.rs [hunk 2]
```diff
@@ -213,24 +213,30 @@ impl BorrowMut<Collab> for DatabaseRow {
...
 
-  pub fn create(row_id: RowId, collab: &mut Collab, row: Row) -> Self {
-    Self::create_with_data(row_id, collab, Some(row))
+  pub fn create(collab: &mut Collab, row: Row) -> Result<Self, DatabaseError> {
+    Self::create_with_data(collab, Some(row))
   }
 
-  fn create_with_data(row_id: RowId, collab: &mut Collab, row: Option<Row>) -> Self {
+  fn create_with_data(collab: &mut Collab, row: Option<Row>) -> Result<Self, DatabaseError> {
+    let row_id: String = collab
```
Comment (appflowy) [inline collab-database/src/rows/row.rs:234]: when create a row, the database row_id must be empty

File: collab-database/src/blocks/block.rs [hunk 3]
```diff
@@ -122,11 +123,15 @@ impl Block {
     }
 
     let encoded_collab = default_database_row_data(&row_id, row);
-    let collab = self
+    let mut collab = self
       .create_collab_for_row(&row_id, Some(encoded_collab))
       .await?;
+    let _ = collab.data.insert_with_path(
```
Comment (appflowy) [inline collab-database/src/blocks/block.rs:129]: there is a row id in meta
image

File: collab-entity/src/proto/collab.rs [hunk 4]
```diff
@@ -1,32 +1,4 @@
 // This file is @generated by prost-build.
-/// Identifier of an active collab document sent over pubsub
-#[allow(clippy::derive_partial_eq_without_eq)]
```
Comment (appflowy) [inline collab-entity/src/proto/collab.rs:3]: why do you modify these?

Comment (speed2exe) [reply collab-entity/src/proto/collab.rs:3]: this is auto generated

File: collab-database/src/rows/row.rs (hunk 2 above)
Comment (speed2exe) [reply collab-database/src/rows/row.rs]: reverted

File: collab-entity/src/proto/collab.rs (hunk 4 above)
Comment (appflowy) [reply collab-entity/src/proto/collab.rs:3]: it seems like auto format. try not to change it. After you change it, i build, then they revert back

File: collab-database/src/rows/row.rs [hunk 5]
```diff
@@ -457,23 +528,40 @@ impl<'a, 'b> RowUpdate<'a, 'b> {
...
-  pub fn set_row_id(self, new_row_id: RowId, database_id: String) -> Self {
-    let old_row_meta = row_id_from_map_ref(self.txn, &self.map_ref)
-      .and_then(|row_id| row_id.parse::<Uuid>().ok())
-      .map(|row_id| RowMeta::from_map_ref(self.txn, &row_id, &self.meta_ref));
+  pub fn set_database_id(self, database_id: String) -> Self {
+    self.map_ref.insert(self.txn, ROW_DATABASE_ID, database_id);
+    self
+  }
 
+  pub fn set_row_id(self, new_row_id: RowId) -> Self {
```
Comment (appflowy) [inline collab-database/src/rows/row.rs:536]: you should update the RowUpdate's  `set_row_id`

**Pull request merged**